exporters/trace/gcp_exporter/internal/recordable.cc

```diff
@@ -0,0 +1,110 @@
+#include "exporters/trace/gcp_exporter/recordable.h"
+#include <assert.h>
+
+
+constexpr char kProjectsPathStr[] = "projects/";
+constexpr char kTracesPathStr[] = "/traces/";
+constexpr char kSpansPathStr[] = "/spans/";
+constexpr char kGCPEnvVar[] = "GOOGLE_CLOUD_PROJECT_ID";
+
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace gcp
+{
+
+void Recordable::SetIds(trace::TraceId trace_id,
+                        trace::SpanId span_id,
+                        trace::SpanId parent_span_id) noexcept
+{
+    std::array<char, 2*trace::TraceId::kSize> hex_trace_buf; 
+    trace_id.ToLowerBase16(hex_trace_buf);
+    const std::string hex_trace(hex_trace_buf.data(), trace::TraceId::kSize);
+
+    std::array<char, 2*trace::SpanId::kSize> hex_span_buf; 
+    span_id.ToLowerBase16(hex_span_buf);
+    const std::string hex_span(hex_span_buf.data(), trace::SpanId::kSize);
+
+    std::array<char, 2*trace::SpanId::kSize> hex_parent_span_buf; 
+    parent_span_id.ToLowerBase16(hex_parent_span_buf);
+    const std::string hex_parent_span(hex_parent_span_buf.data(), trace::SpanId::kSize);
+
+    // Get Project Id
+    const std::string project_id(getenv(kGCPEnvVar));
+    
+    span_.set_name(kProjectsPathStr + project_id + kTracesPathStr + hex_trace + kSpansPathStr + hex_span);
+    span_.set_span_id(hex_span);
+    span_.set_parent_span_id(hex_parent_span);
+}
+
+
+void Recordable::SetAttribute(nostd::string_view key,
+                              const common::AttributeValue &&value) noexcept
+{
+    assert(nostd::holds_alternative<bool>(value) ||
+           nostd::holds_alternative<int64_t>(value) ||
+           nostd::holds_alternative<nostd::string_view>(value));
+
```

Comment:
Based on https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#basic-error-handling-principles, API can only fail on initialization, but shouldn't make application fail at runtime.

Because of that, there should be no asserts in recordable/exporter code outside of constructors/initialization code.

Also, common::AttributeValue can legitimately hold types other than 3 types listed here based on definition in https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/common/attribute_value.h 

We should handle all those types here eventually (say all int types can be converted to int value in the map, and 
we need to decide on how to handle arrays properly). We can add more types as a followup in the future PR.

Reply:
Yes, this was concerning to me too. In Opencensus, they assume that always the correct type is passed as an attribute value, so I thought there should be some check in place for that in OT.